### PR TITLE
 Added check for result correctness for Geometric functions

### PR DESCRIPTION
**Pull request by kopylovanat**
Also covered previously missed gap - type double for geometric functions

Depends on PR https://github.com/KhronosGroup/SYCL-CTS/pull/78
Only last commit https://github.com/KhronosGroup/SYCL-CTS/commit/5a87c69a21282f719857fe919b744bcefa88a452 for this PR

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

**Review by keryell — CHANGES_REQUESTED** [review]
That looks good.
Just a few details.

File: util/math_reference.cpp [hunk 1]
```diff
@@ -1025,4 +1032,482 @@ cl::sycl::uint16 mul24(cl::sycl::uint16 x, cl::sycl::uint16 y) {
...
+
+  union {
+    int16_t i;
+    cl::sycl::half f;
+  } a, b;
+
+  a.f = x;
+  b.f = y;
+
+  if (a.i & 0x8000)
```
[the remark this answers is not in the document]
Comment (kopylovanat) [reply util/math_reference.cpp]: Added comments.

File: tests/math_builtin_api/modules/sycl_functions.py [hunk 2]
```diff
@@ -174,49 +176,49 @@ def create_common_signatures():
 def create_geometric_signatures():
     sig_list = []
 
-    f_cross = funsig("cl::sycl", "cl::sycl::float3", "cross", ["cl::sycl::float3", "cl::sycl::float3"])
-    sig_list.append(f_cross)
+    f_cross = funsig("cl::sycl", "cl::sycl::float3", "cross", ["cl::sycl::float3", "cl::sycl::float3"], "3") # cumulative error for multiplications and substruction
```
[the remark this answers is not in the document]
Comment (kopylovanat) [reply tests/math_builtin_api/modules/sycl_functions.py]: Added displaying accuracy comments if something goes wrong

File: util/math_reference.cpp [hunk 3]
```diff
@@ -1275,23 +1275,20 @@ cl::sycl::half nextafter(cl::sycl::half x, cl::sycl::half y) {
...
+    a = 0x8000 - a;
+  if (b & 0x8000)
+    b = 0x8000 - b;
 
-  if (a.i & 0x8000)
-    a.i = 0x8000 - a.i;
-  if (b.i & 0x8000)
-    b.i = 0x8000 - b.i;
+  a += (a < b) ? 1 : -1;
+  a = (a < 0) ? (int16_t)0x8000 - a : a;
```
Comment (keryell) [inline util/math_reference.cpp:1289]: Not clear what this is doing. Perhaps something like:
`// Convert again a from 2-complement signed value into sign bit + unsigned value`

**Review by keryell — CHANGES_REQUESTED** [review]
Thanks, quite clearer.
There are still some obscure details...

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**